Source/AwsCommonRuntimeKit/io/StreamCore.swift

```diff
@@ -0,0 +1,93 @@
+//  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//  SPDX-License-Identifier: Apache-2.0.
+
+import AwsCIo
+
+class AwsInputStreamCore {
+    var rawValue: aws_input_stream
+    let awsStream: AwsStream
+    let vtable: AwsInputStreamVtable
+    init(awsStream: AwsStream, allocator: Allocator) {
+        self.awsStream = awsStream
+        rawValue = aws_input_stream()
+        self.vtable = AwsInputStreamVtable(allocator: allocator)
+        rawValue.vtable = vtable.rawValue
+        rawValue.impl = Unmanaged<AwsInputStreamCore>.passUnretained(self).toOpaque()
+    }
+}
+
+class AwsInputStreamVtable: CStruct {
+    private let _rawValue: UnsafeMutablePointer<aws_input_stream_vtable>
+    var rawValue: UnsafePointer<aws_input_stream_vtable> { UnsafePointer(_rawValue) }
+    let allocator: Allocator
+    init(allocator: Allocator) {
+        self.allocator = allocator
+        _rawValue = allocator.allocate(capacity: 1)
+        _rawValue.pointee.seek = doSeek
+        _rawValue.pointee.read = doRead
+        _rawValue.pointee.get_status = doGetStatus
+        _rawValue.pointee.get_length = doGetLength
+        _rawValue.pointee.acquire = {
+            _ = Unmanaged<AwsInputStreamCore>.fromOpaque($0!.pointee.impl).retain()
+        }
+        _rawValue.pointee.release = {
+            Unmanaged<AwsInputStreamCore>.fromOpaque($0!.pointee.impl).release()
+        }
+    }
+
+    typealias RawType = aws_input_stream_vtable
+    func withCStruct<Result>(_ body: (aws_input_stream_vtable) -> Result) -> Result {
+        return body(_rawValue.pointee)
+    }
+
+    deinit {
+        allocator.release(_rawValue)
+    }
+}
```

Comment:
I don’t like this, however, I can’t think of a better way. Previous implementation had undefined behavior as you are not supposed to store pointer by &.